.github/workflows/ci.yml

```diff
@@ -603,8 +624,13 @@ ${{ hashFiles('.pre-commit-config.yaml') }}"
         uses: actions/setup-python@v2
         with:
           python-version: ${{ env.PYTHON_MAJOR_MINOR_VERSION }}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+          cache: 'pip'
+      - run: python -m pip install --editable ../../dev/breeze/
```

Comment:
How does this behave with having two python version set up? Will this possibly cause some of the tests to now only use python 3.7?

Reply:
No relation wth that. Our tests will continue using the images (controlled by PYTHON_MAJOR_MINOR_VERSION) variable.

This one is the "Breeze" virtualenv only that has nothing to do with the "airflow" virtualenv. It's only there to run the CI and ./Breeze scripts (it's basically a "common" execution virtualenv that is portable way of running our scripts in any OS - replacing what `Bash` was supposed to do in the original "Breeze". Just common execution (virtual) environment with all dependencies installed that the scripts are using.

For "users" use, this will be installed either automatically by bootstraping script `./Breeze2` or (whe we stabilize dependencie) via `pipx instal --editable ./dev/breeze`. 

More details and reasoning (alternatives considered) are described in this ADR: https://github.com/apache/airflow/blob/main/dev/breeze/doc/adr/0003-bootstraping-virtual-environment.md

Reply:
Right before this snippet we have a `setup-python@v2` -- do we need both?

Reply:
Yes. The `setup-python@v2' installs python (and sets up PIP cache). Then you need to run `pip' install to install dependencies: See https://github.com/actions/setup-python#caching-packages-dependencies for examples.

Reply:
But we have wrong cache-dependency path :).

Reply:
This I mean:

```
      - name: "Setup python"
        uses: actions/setup-python@v2
        with:
          python-version: ${{ env.PYTHON_MAJOR_MINOR_VERSION }}
      - uses: actions/setup-python@v2
        with:
          python-version: '3.7'
          cache: 'pip'
```

We have two `actions/setup-python@v2` -- it seems the first one is "overwritten" by the second one (in terms of what `python` would be at run time.)

Reply:
Ah - yeah that is a merge artifact. @edithturn had some trouble with rebasing so this is likely result of that.

Reply:
There should be only one, indeed.